### PR TITLE
Include a commit with K8s 1.20 libs

### PR DESCRIPTION
**Pull request by dprotaso**
I'm trying to bump Knative's K8s dependency to 1.20 - but this repo skipped 1.20 to 1.21.

I'm hoping this PR is a noop for most - but gives me the ability to target a specific commit that uses K8s 1.20.x deps.

Thus I'm expecting the commit history to _not_ be squashed

### CODE REVIEW

**Comment by dprotaso** [comment]
Is this a flakey test?

> progress_test.go:227: saw progress revert: was high of 6726222, saw 6660686

**Comment by imjasonh** [comment]
> Is this a flakey test?

Yep 😢 

This change sgtm

File: pkg/authn/k8schain/go.mod [hunk 1]
```diff
@@ -3,15 +3,27 @@ module github.com/google/go-containerregistry/pkg/authn/k8schain
 go 1.14
 
 require (
+	github.com/docker/spdystream v0.0.0-20160310174837-449fdfce4d96 // indirect
+	github.com/fsnotify/fsnotify v1.4.9 // indirect
```
Comment (jonjohnsonjr) [inline pkg/authn/k8schain/go.mod:7]: 🤔

Comment (dprotaso) [reply pkg/authn/k8schain/go.mod:7]: yeah let me re-run the commands

Comment (dprotaso) [reply pkg/authn/k8schain/go.mod:7]: https://github.com/kubernetes/apimachinery/blob/v0.21.1/go.mod#L21

fsnotify is from gomega which k8s uses

Comment (dprotaso) [reply pkg/authn/k8schain/go.mod:7]: it's not obvious to me why the go tool chain made it show up as `//indirect` here

Comment (dprotaso) [reply pkg/authn/k8schain/go.mod:6]: `go mod tidy` FTW

Comment (dprotaso) [reply pkg/authn/k8schain/go.mod]: fsnotify remained - I deleted all `//indirect` dependencies and ran `go mod tidy` again

**Review by jonjohnsonjr — APPROVED** [review]

**Comment by dprotaso** [comment]
hmm.. did you squash this commit?

**Comment by jonjohnsonjr** [comment]
See green button. Push green button. 🐒

**Pull request merged**